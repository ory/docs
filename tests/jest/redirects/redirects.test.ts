const { pathToRegexp, match, parse, compile } = require('path-to-regexp')

it('should ensure the regex is ok', () => {
  const regexp = pathToRegexp('/docs/hydra/:version(v[0-9\\.]+|next)/:path*')

  expect(regexp.exec('/docs/hydra/v1.9/foo')).toMatchSnapshot()
  expect(regexp.exec('/docs/hydra/v1.9/')).toMatchSnapshot()
  expect(regexp.exec('/docs/hydra/v1.9')).toMatchSnapshot()
})
