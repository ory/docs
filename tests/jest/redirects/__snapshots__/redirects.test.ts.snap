// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should ensure the regex is ok 1`] = `
Array [
  "/docs/hydra/v1.9/foo",
  "v1.9",
  "foo",
]
`;

exports[`should ensure the regex is ok 2`] = `
Array [
  "/docs/hydra/v1.9/",
  "v1.9",
  undefined,
]
`;

exports[`should ensure the regex is ok 3`] = `
Array [
  "/docs/hydra/v1.9",
  "v1.9",
  undefined,
]
`;
